File: src/android/APIService.java
package cordova.plugin.prosegur.geolocation;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.Header;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface APIService {

    @POST("/GeoLocation_API/rest/DeviceGeoLocations/SetDeviceGeoLocations")
    Call<GeoPosittion> sendPostGeo(@Header("Token") String token, @Query("country") String country, @Query("imei") String imei, @Body ArrayList<GeoPosittion> body);

}

File: src/android/APIService.kt
package cordova.plugin.prosegur.geolocation

import retrofit2.http.Body
import retrofit2.http.Header
import retrofit2.http.POST
import retrofit2.http.Query
import retrofit2.http.Url

interface APIService {

    @POST
    suspend fun sendPostGeo(
        @Url fullUrl: String,
        @Header("Token") token: String,
        @Query("country") country: String,
        @Query("imei") imei: String,
        @Body body: ArrayList<GeoPosittion> = arrayListOf()
    ): GeoPosittion
}